src/analysis/souffle/taint.dl

```diff
@@ -61,6 +69,9 @@
 // There is a claim that `accessPath` definitely has `tag` taint.
 .decl claimHasTag(accessPath: AccessPath, tag: Tag)
 
+// There is a check that `accessPath` has a particular presence of `tag` taint.
+.decl checkTag(accessPath: AccessPath, tag: Tag, tagPresence: TagPresence)
```

Comment:
A check can be a complicated boolean expression. Therefore, in all likelihood, we will convert a `check` into inference rules. In the meantime, if you need this for incremental progress, let us add something like `checkTagIsPresent` and `checkTagIsAbsent` instead of the `tagPresence` argument?

Reply:
OK, sounds good, I can do that.

Reply:
Done. Only `checkTagIsPresent` is needed for the tests in this PR, so I have simplified to only include that relation for now.